### ISSUE TITLE
v2 issue: TLS Error: Certificate verification failed

### ISSUE THREAD

**Issue by Captain-Barge** [yuezk/GlobalProtect-openconnect#295]
Updated to the new GlobalProtect today (v2.0.0-beta5) and now I can't connect. Been using GlobalProtect for a long time without issues so I believe the issue is with the new version.

Error message appears after I enter my username/password/2FA:
![Screenshot from 2024-01-22 13-54-25](https://github.com/yuezk/GlobalProtect-openconnect/assets/83183390/9f2e6fe6-bf88-4b13-9537-0593154ee4e9)

Setting `Ignore TLS Errors` in the settings (& restarting the app) makes no difference.
![image](https://github.com/yuezk/GlobalProtect-openconnect/assets/83183390/b9039f54-b906-4797-ba5a-fc40690103da)

I tried running from the terminal as well with:
`gpclient --ignore-tls-errors --fix-openssl connect vpn.xyxyxyxy.com:123`

This produces the error message:

> Error: error sending request for url (https://vpn.xyxyxyxy.com/global-protect/getconfig.esp): error trying to connect: error:0A000086:SSL routines:tls_post_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:2092: (unable to get local issuer certificate)
> 
> Re-run it with the `--ignore-tls-errors` option to ignore the certificate error, e.g.:
> 
> gpclient --ignore-tls-errors --ignore-tls-errors --fix-openssl connect vpn.xyxyxyxy.com:123

Note that the error message suggests adding the `--ignore-tls-errors` CLI option twice (which is non-sensical).

If it's helpful, below is the full terminal output:

> [2024-01-22T20:55:32Z INFO  gpclient::cli] gpclient started: 2.0.0-beta5 (2024-01-22)
> [2024-01-22T20:55:32Z INFO  gpclient::cli] TLS errors will be ignored
> [2024-01-22T20:55:32Z INFO  gpapi::portal::prelogin] Portal prelogin, user_agent: PAN GlobalProtect
> [2024-01-22T20:55:33Z INFO  gpauth::cli] gpauth started: 2.0.0-beta5 (2024-01-22)
> [2024-01-22T20:55:33Z INFO  gpauth::cli] TLS errors will be ignored
> [2024-01-22T20:55:33Z INFO  gpauth::cli] Fixing OpenSSL environment
> [2024-01-22T20:55:33Z INFO  gpauth::auth_window] Open auth window, user_agent: PAN GlobalProtect
> [2024-01-22T20:55:33Z INFO  gpauth::auth_window] Auth window user agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Safari/605.1.15
> [2024-01-22T20:55:33Z INFO  gpauth::auth_window] Load the SAML request as URI...
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] Loaded uri: https://a**********m/login/?authkey=A**********5&scid=b**********5
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] No saml-auth-status header found
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] No auth data found in headers, trying to read from body...
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] No auth data found in HTML
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] No auth data found, it may not be the /SAML20/SP/ACS endpoint
> [2024-01-22T20:55:34Z INFO  gpauth::auth_window] Raise window in 1 second(s)
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] Loaded uri: https://a**********m/frame/frameless/v4/auth?sid=f**********4&tx=e**********w&req-trace-group=0**********f
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] No saml-auth-status header found
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] No auth data found in headers, trying to read from body...
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] No auth data found in HTML
> [2024-01-22T20:55:41Z INFO  gpauth::auth_window] No auth data found, it may not be the /SAML20/SP/ACS endpoint
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] Loaded uri: https://a**********m/frame/v4/preauth/healthcheck?sid=f**********4
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No saml-auth-status header found
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found in headers, trying to read from body...
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found in HTML
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found, it may not be the /SAML20/SP/ACS endpoint
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] Loaded uri: https://a**********m/frame/frameless/v4/auth?sid=f**********4&tx=e**********w
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No saml-auth-status header found
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found in headers, trying to read from body...
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found in HTML
> [2024-01-22T20:55:42Z INFO  gpauth::auth_window] No auth data found, it may not be the /SAML20/SP/ACS endpoint
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] Loaded uri: https://a**********m/duo/ASI2F9QNRR8ZW87MIXVS/callback?state=p**********5&duo_code=J**********I
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] No saml-auth-status header found
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] No auth data found in headers, trying to read from body...
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] No auth data found in HTML
> [2024-01-22T20:55:43Z INFO  gpauth::auth_window] No auth data found, it may not be the /SAML20/SP/ACS endpoint
> [2024-01-22T20:55:44Z INFO  gpauth::auth_window] Loaded uri: https://v**********m/SAML20/SP/ACS
> [2024-01-22T20:55:44Z INFO  gpauth::auth_window] Trying to read auth data from response headers...
> [2024-01-22T20:55:44Z INFO  gpauth::auth_window] Got auth data from headers
> [2024-01-22T20:55:44Z INFO  gpapi::portal::config] Portal config, user_agent: PAN GlobalProtect
> 
> Error: error sending request for url (https://vpn.xyxyxyxy.com/global-protect/getconfig.esp): error trying to connect: error:0A000086:SSL routines:tls_post_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:2092: (unable to get local issuer certificate)
> 
> Re-run it with the `--ignore-tls-errors` option to ignore the certificate error, e.g.:
> 
> gpclient --ignore-tls-errors --ignore-tls-errors --fix-openssl connect vpn.xyxyxyxy.com:123
> 

I'm on Ubuntu 22.04.3

**Comment by Captain-Barge** [comment]
I'll add that in the pre-v2 version of GlobalProtect I had added `openconnect-args=--servercert pin-sha256:xxxxxxxxxxxxxxxxxxxxxxxxxxx` to my config file at `/etc/gpservice/gp.conf`. This was necessary to get the old version working.

I think I see that the new version has added a `gp.conf` file located at `/etc/gp.conf`. I tried updating that one with the line above but with no success.

Maybe none of that is relevant, but I thought I'd mention.
